rfcs/2020-08-26-3191-host-metrics.md

```diff
@@ -0,0 +1,157 @@
+- `memory_free_bytes` (gauge)
+- `memory_inactive_bytes` (gauge)
+- `memory_swap_total_bytes` (gauge)
+- `memory_swap_used_bytes` (gauge)
+- `memory_swapped_in_bytes_total` (gauge)
+- `memory_swapped_out_bytes_total` (gauge)
+- `memory_total_bytes` (gauge)
+- `memory_wired_bytes` (gauge)
+
+Users should be able to limit the collection of metrics to specific classes, here: `cpu`, `memory`, `disk/filesystem`, `load`.
+
+Metrics will also be labeled with:
+
+- `host`: the host name of the host being monitored.
+
+## Doc-level Proposal
+
+The following additional source configuration will be added:
+
+```toml
+[sources.my_source_id]
+  type = "host_metrics" # required
+  collecting = [ "all"] # optional, defaults collecting all metrics.
+  scrape_interval_secs = 15 # optional, default, seconds
+  namespace = "aspace" # optional, default, namespace to put metrics under
+```
+
+For `collecting` (name open to discussion) we'd specify an array of metric collections:
+
+```toml
+collecting = [ "cpu", "memory" ]
+```
+
+## Rationale
+
+CPU, Memory, and Disk are the basic building blocks of host-based monitoring. They are considered "table stakes" for most metrics-based monitoring of hosts. Additionally, if we do not support ingesting metrics from them, it is likely to push people to use another tool
+
+As part of Vector's vision to be the "one tool" for ingesting and shipping
+observability data, it makes sense to add as many sources as possible to reduce
+the likelihood that a user will not be able to ingest metrics from their tools.
+
+## Prior Art
+
+- https://github.com/influxdata/telegraf/tree/master/plugins/inputs/cpu
+- https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mem
+- https://github.com/influxdata/telegraf/tree/master/plugins/inputs/disk
+- https://github.com/elastic/beats/tree/master/metricbeat
+- https://github.com/prometheus/node_exporter
+
+## Drawbacks
+
+- Additional maintenance and integration testing burden of a new source
+
+## Alternatives
+
+### Having users run telegraf or Prom node exporter and using Vector's prometheus source to scrape it
+
+We could not add the source directly to Vector and instead instruct users to run
+Telegraf or Prometheus' node exporter and point Vector at the exposed Prometheus scrape endpoint. This would leverage the already supported inputs from those projects.
+
+I decided against this as it would be in contrast with one of the listed
+principles of Vector:
+
+> One Tool. All Data. - One simple tool gets your logs, metrics, and traces
+> (coming soon) from A to B.
+
+[Vector
+principles](https://vector.dev/docs/about/what-is-vector/#who-should-use-vector)
+
+On the same page, it is mentioned that Vector should be a replacement for
+Telegraf.
+
+> You SHOULD use Vector to replace Logstash, Fluent*, Telegraf, Beats, or
+> similar tools.
+
+If users are already running Telegraf or Node Exporter though, they could opt for this path.
+
+## Outstanding Questions
```

Comment:
This isn't really an outstanding question, but just noting that we'll need to take consideration for documenting how to use this source within a `docker` container (similar to https://github.com/prometheus/node_exporter#using-docker).